src/app/features/add-network/add-network.tsx

```diff
@@ -36,84 +36,163 @@ interface AddNetworkFormValues {
+      // The `.parent_network_id` refers to the actual peer network ID in both cases
+      const { network_id: chainId, parent_network_id: parentNetworkId } = chainInfo;
+
+      const isSubnet = typeof chainInfo.l1_subnet_governing_contract === 'string';
+      const isFirstLevelSubnet =
+        isSubnet &&
+        (parentNetworkId === PeerNetworkID.Mainnet || parentNetworkId === PeerNetworkID.Testnet);
+
+      // Currently, only subnets of mainnet and testnet are supported in the wallet
+      if (isFirstLevelSubnet) {
+        const parentChainId =
+          parentNetworkId === PeerNetworkID.Mainnet ? ChainID.Mainnet : ChainID.Testnet;
+        networksActions.addNetwork({
+          chainId: parentChainId, // Used for differentiating control flow in the wallet
+          subnetChainId: chainId, // Used for signing transactions (via the network object, not to be confused with the NetworkConfigurations)
+          id: key as DefaultNetworkConfigurations,
+          name,
+          url: path,
+        });
+        navigate(RouteUrls.Home);
+        return;
+      }
+
+      if (chainId === ChainID.Mainnet || chainId === ChainID.Testnet) {
+        networksActions.addNetwork({
+          chainId,
+          id: key as DefaultNetworkConfigurations,
+          name,
+          url: path,
+        });
+        navigate(RouteUrls.Home);
+        return;
+      }
+
+      setError('Unable to determine chainID from node.');
+    } catch (error) {
+      setError('Unable to fetch info from node.');
+    }
+  };
+
+  const setCustomBitcoinNetwork = async (values: AddNetworkFormValues) => {
+    try {
+      const { name, url } = values;
+      const path = removeTrailingSlash(new URL(url).href);
+
+      networksActions.addNetwork({
+        chainId: ChainID.Testnet,
+        id: 'bitcoin-regtest' as DefaultNetworkConfigurations,
+        name,
+        url: path,
+      });
+      navigate(RouteUrls.Home);
+      return;
+    } catch (error) {
+      setError('Unable to fetch info from node.');
+    }
+  };
+
   return (
     <CenteredPageContainer>
+      <Stack
+        maxWidth={token('sizes.centeredPageFullWidth')}
+        px={['loose', 'base-loose']}
+        spacing="loose"
+        textAlign={['left', 'center']}
+        padding={'loose'}
+      >
+        <Text>
+          Use this form to add a new instance of the{' '}
+          <a
+            href="https://github.com/blockstack/stacks-blockchain-api"
+            target="_blank"
+            rel="noreferrer"
+          >
+            Stacks Blockchain API
+          </a>{' '}
+          or{' '}
+          <a href="https://github.com/Blockstream/esplora" target="_blank" rel="noreferrer">
+            Bitcoin Blockchain API
+          </a>
+          . Make sure you review and trust the host before you add it.
+        </Text>
+        <Stack justifyContent={'center'}>
+          <LeatherButton
+            onClick={() => setBlockchain('stacks')}
+            backgroundColor={
```

Comment:
Can you post a screenshot of these UI changes? There are `variants` of the LeatherButton to use here.

Reply:
These would better be tabs than buttons. You can reuse the tabs from the homepage

Reply:
Although, wouldn't it be better to have this all on the same page, seeing as you need them all?

Reply:
So these buttons won't take you to another page, it just changes the blockchain variable. So if you click Bitcoin, you'll add a Bitcoin network with this form.